mesonbuild/compilers/mixins/clike.py

```diff
@@ -925,13 +926,83 @@ def symbols_have_underscore_prefix(self, env: 'Environment') -> bool:
                     # Check if the underscore form of the symbol is somewhere
                     # in the output file.
                     if b'_' + symbol_name in line:
-                        mlog.debug("Symbols have underscore prefix: YES")
+                        mlog.debug("Underscore prefix check found prefixed function in binary")
                         return True
                     # Else, check if the non-underscored form is present
                     elif symbol_name in line:
-                        mlog.debug("Symbols have underscore prefix: NO")
+                        mlog.debug("Underscore prefix check found non-prefixed function in binary")
                         return False
-        raise RuntimeError(f'BUG: {n!r} check failed unexpectedly')
+        return None
```

Comment:
This is always the last check, and if I understand correctly, it's a Meson bug if this doesn't have one or the other, right?

So I don't think it should return None.

Reply:
Indeed, will re-add this.